src/Microsoft.AspNetCore.Razor.Language/RazorProjectEngine.cs

```diff
@@ -20,60 +20,45 @@ public abstract class RazorProjectEngine
 
-        public static RazorProjectEngine Create(RazorProjectFileSystem fileSystem, Action<RazorProjectEngineBuilder> configure)
+        public static RazorProjectEngine Create(RazorProjectFileSystem fileSystem, Action<RazorProjectEngineBuilder> configure) => Create(fileSystem, configure, RazorConfiguration.Default);
+
+        public static RazorProjectEngine Create(
+            RazorProjectFileSystem fileSystem,
+            Action<RazorProjectEngineBuilder> configure,
+            RazorConfiguration configuration)
         {
             if (fileSystem == null)
             {
                 throw new ArgumentNullException(nameof(fileSystem));
             }
 
-            var builder = new DefaultRazorProjectEngineBuilder(designTime: false, fileSystem: fileSystem);
-
-            AddDefaults(builder);
-            AddRuntimeDefaults(builder);
-            configure?.Invoke(builder);
-
-            return builder.Build();
-        }
-
-        public static RazorProjectEngine CreateDesignTime(RazorProjectFileSystem fileSystem) => CreateDesignTime(fileSystem, configure: null);
-
-        public static RazorProjectEngine CreateDesignTime(RazorProjectFileSystem fileSystem, Action<RazorProjectEngineBuilder> configure)
-        {
-            if (fileSystem == null)
+            if (configuration == null)
             {
-                throw new ArgumentNullException(nameof(fileSystem));
+                throw new ArgumentNullException(nameof(configuration));
             }
 
-            var builder = new DefaultRazorProjectEngineBuilder(designTime: true, fileSystem: fileSystem);
+            var builder = new DefaultRazorProjectEngineBuilder(configuration, fileSystem);
 
             AddDefaults(builder);
-            AddDesignTimeDefaults(builder);
-            configure?.Invoke(builder);
 
-            return builder.Build();
-        }
-
-        public static RazorProjectEngine CreateEmpty(RazorProjectFileSystem fileSystem, Action<RazorProjectEngineBuilder> configure)
-        {
-            if (fileSystem == null)
+            if (configuration.DesignTime)
             {
-                throw new ArgumentNullException(nameof(fileSystem));
+                AddDesignTimeDefaults(builder);
             }
-
-            if (configure == null)
+            else
             {
-                throw new ArgumentNullException(nameof(configure));
+                AddRuntimeDefaults(builder);
             }
 
-            var builder = new DefaultRazorProjectEngineBuilder(designTime: false, fileSystem: fileSystem);
-
-            configure(builder);
+            configure?.Invoke(builder);
 
             return builder.Build();
         }
 
-        public static RazorProjectEngine CreateDesignTimeEmpty(RazorProjectFileSystem fileSystem, Action<RazorProjectEngineBuilder> configure)
+        public static RazorProjectEngine CreateEmpty(
```

Comment:
Do we have a strong use case for this as a public API?

Reply:
Not today. Can always add later, i'll nuke it.